src/views/capture-view.ts

```diff
@@ -109,46 +111,72 @@ export default class CaptureView extends View {
       const currentIndex = devices.indexOf(this.currentDevice);
 
       // TODO: Should show chooser for many devices, not just cycle
-      this.currentDevice = devices[(currentIndex + 1) % devices.length];
+      this.chooseCamera(devices[(currentIndex + 1) % devices.length]);
     } else {
-      this.currentDevice = devices[0];
+      this.chooseCamera(devices[0]);
     }
-
-    this.startStream(this.currentDevice.deviceId);
   }
 
   close() {
     router.visit(`/browse`);
   }
 
+  private async chooseCamera(camera: MediaDeviceInfo) {
+    this.currentDevice = camera;
+    await this.startStream(this.currentDevice.deviceId);
+    const trackCapabilities = this.cameraHelper.getCapabilities();
```

Comment:
You may want to remove `trackCapabilities` from there and `getCapabilities` and `setConstraint` from camera-helper.js as they're not used.